Sources/SwiftMemcache/MemcachedConnection.swift

```diff
@@ -171,19 +183,111 @@ public actor MemcachedConnection {
         }
     }
 
-    /// Set the value for a key on the Memcache server.
+    // MARK: - Updating Time-To-Live
+
+    /// Update the TTL for a key in the Memcache server.
+    ///
+    /// - Parameters:
+    ///   - key: The key for which to update the TTL.
+    ///   - newTimeToLive: The new Time-To-Live value. If `nil`, the TTL will not be updated.
+    /// - Throws: A `MemcachedConnectionError` if the connection is shutdown or if there's an unexpected nil response.
+    public func get<Value: MemcachedValue>(_ key: String, as valueType: Value.Type = Value.self, newTimeToLive: TimeToLive? = nil) async throws {
+        switch self.state {
+        case .initial(_, _, _, let requestContinuation, let clock),
+             .running(_, _, _, let requestContinuation, let clock):
+
+            var flags = MemcachedFlags()
+            if let newTimeToLive {
+                flags = MemcachedFlags()
+                flags.timeToLive = newTimeToLive.durationUntilExpiration(inRelationTo: clock)
+            }
+            flags.shouldReturnValue = true
+
+            let command = MemcachedRequest.GetCommand(key: key, flags: flags)
+            let request = MemcachedRequest.get(command)
+
+            _ = try await withCheckedThrowingContinuation { continuation in
+                switch requestContinuation.yield((request, continuation)) {
+                case .enqueued:
+                    break
+                case .dropped, .terminated:
+                    continuation.resume(throwing: MemcachedConnectionError.connectionShutdown)
+                default:
+                    break
+                }
+            }
+
+        case .finished:
+            throw MemcachedConnectionError.connectionShutdown
+        }
```

Comment:
We have been duplicating this code a bunch now. We should write a `private` method that takes a `MemcacheRequest` and does the continuation dance for us so that we just have to create the request and get the correct response.